script.js

```diff
@@ -4,7 +4,13 @@ import fs from 'node:fs/promises'
 const base = await fs.readFile(path.join('test', 'core.js'))
 
 const lines = String(base)
-  .replace(/\bresolve(?=\(|,)/g, 'import.meta.resolve')
+  .replace(/\bresolve(?=\()/g, 'await import.meta.resolve')
+  .replace(/\bresolve(?=,)/g, 'import.meta.resolve')
+  .replace(
+    /const execute = .*$/g,
+    'const execute = async (/** @type {() => Promise<void>} */ f) => f()'
+  )
+  .replace(/execute\(/g, 'await execute(async ')
   .split('\n')
```

Comment:
Should we perhaps generate two baseline test files, one async one sync, and a runtime file that checks the current Node version, and imports one or the other?

Reply:
I actually thought of doing this, but there is currently _no_ version of Node.js that supports the sync version! I will try and remember, once one comes out, of doing this. And also cleaning up some out of date code that seems to deal with _really_ old versions of Node.js (I _think_ I saw something there).